config/config-api/src/com/thoughtworks/go/config/EnvironmentVariableConfig.java

```diff
@@ -339,4 +348,18 @@ public void setEntityId(Long entityId) {
     public void setEntityType(String entityType) {
         this.entityType = entityType;
     }
+
+    @Override
+    public ConfigOrigin getOrigin() {
+        return origin;
+    }
+
+    public boolean isRemote()
+    {
+        return origin != null && !origin.isLocal();
```

Comment:
There are similar checks implemented at few other places. Does it make sense to implement a null object pattern here like `NullOrigin` instead of checking for not-null and then for `isLocal()` or `isRemote` everywhere?